sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java

```diff
@@ -0,0 +1,234 @@
+
+            // 4. Create an event grid subscription.
+            EventSubscription eventSubscription = eventGridManager.eventSubscriptions()
+                .define(eventSubscriptionName)
+                .withExistingScope(eventGridTopic.id())
+                .withDestination(new EventHubEventSubscriptionDestination()
+                    .withResourceId(eventHub.id()))
+                .withFilter(new EventSubscriptionFilter()
+                    .withIsSubjectCaseSensitive(false)
+                    .withSubjectBeginsWith("")
+                    .withSubjectEndsWith(""))
+                .create();
+
+            System.out.println("EventGrid event subscription created with name " + eventSubscription.name());
+
+            // 5. Retrieve the event grid client connection key.
+            String eventGridClientConnectionKey = eventGridManager.topics().listSharedAccessKeys(resourceGroupName, topicName).key1();
+
+            System.out.format("Found EventGrid client connection key \"%s\" for endpoint \"%s\"\n", eventGridClientConnectionKey, eventGridTopic.endpoint());
+
+            // 6. Create an event grid publisher client.
+            EventGridPublisherClient<EventGridEvent> eventGridPublisherClient = new EventGridPublisherClientBuilder()
+                .endpoint(eventGridTopic.endpoint())
+                .credential(new AzureKeyCredential(eventGridClientConnectionKey))
+                .buildEventGridEventPublisherClient();
+
+            System.out.println("Done creating event grid publisher client.");
+
+            // 7. Create an event hub consumer client
+            String connectionString = eventHub.listAuthorizationRules().stream().findFirst().get().getKeys().primaryConnectionString();
+            System.out.format("Event hub connection string: %s%n", connectionString);
+
+            EventHubConsumerAsyncClient consumer = new EventHubClientBuilder()
+                .connectionString(connectionString)
+                .consumerGroup(EventHubClientBuilder.DEFAULT_CONSUMER_GROUP_NAME)
+                .buildAsyncConsumerClient();
+
+            System.out.println("Done creating event hub consumer client.");
+
+            // 8. Subscribe to coming events from event grid using `EventHubConsumerAsyncClient`.
+            // 8.1 Get the first partition and block on it
+            String firstPartition = consumer.getPartitionIds().blockFirst(OPERATION_TIMEOUT);
+
+            // This shouldn't happen, but if we are unable to get the partitions within the timeout period.
+            if (firstPartition == null) {
+                firstPartition = "0";
+            }
+
+            // 8.2 Subscribe to the coming events from event grid
+            // We use CountDownLatch to wait for the coming events
+            CountDownLatch countDownLatch = new CountDownLatch(NUMBER_OF_EVENTS);
+            consumer.receiveFromPartition(firstPartition, EventPosition.latest())
+                .subscribe(partitionEvent -> {
```

Comment:
Ok, `.subscribe` is fine. This part actually happen after the `sendEvent`. You cannot block here.

Reply:
This consuming part is referred to this sample: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/eventhubs/azure-messaging-eventhubs/src/samples/java/com/azure/messaging/eventhubs/ConsumeEvents.java